packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -1568,7 +1568,17 @@ class PopupMenuButton<T> extends StatefulWidget {
 /// of your button state.
 class PopupMenuButtonState<T> extends State<PopupMenuButton<T>> {
   bool _isMenuExpanded = false;
+  RelativeRect? _lastPosition;
+
   RelativeRect _positionBuilder(BuildContext _, BoxConstraints constraints) {
+    if (!mounted) {
+      // When the route is displayed, the `_positionBuilder` closure is stored.
+      // Even after the button has been unmounted and the context becomes invalid,
+      // the route might keep displaying, and `_positionBuilder` must continue to
+      // work in that case.
+      return _lastPosition ?? RelativeRect.fromSize(Rect.zero, constraints.biggest);
+    }
+
     final PopupMenuThemeData popupMenuTheme = PopupMenuTheme.of(context);
     final RenderBox button = context.findRenderObject()! as RenderBox;
```

Comment:
Since this is already implemented this way, so not blocking the merge. but if the goal is to pin the popup to this button with a delta offset, we should use compositedtrasnformfollower and target/leaderlayer. Things will just be simpler with no mounted/unmounted problem

Reply:
@chunhtai Are you talking about https://github.com/flutter/flutter/pull/169814 ?

Reply:
yes, that is right. although looking at the comment it mentioned some bug about using compsitedfollower, which is new to me.